repo/ecc/ecc_rs_bw.go

```diff
@@ -0,0 +1,186 @@
+package ecc
+
+import (
+	"github.com/kopia/kopia/internal/gather"
+	"github.com/kopia/kopia/repo/encryption"
+	"github.com/pkg/errors"
+	"github.com/vivint/infectious"
+)
+
+const (
+	RsBwFs1pEccName  = "RS-BW-FS-1%"
+	RsBwFs2pEccName  = "RS-BW-FS-2%"
+	RsBwFs5pEccName  = "RS-BW-FS-5%"
+	RsBwFs10pEccName = "RS-BW-FS-10%"
+	RsBwFb1pEccName  = "RS-BW-FB-1%"
+	RsBwFb2pEccName  = "RS-BW-FB-2%"
+	RsBwFb5pEccName  = "RS-BW-FB-5%"
+	RsBwFb10pEccName = "RS-BW-FB-10%"
+)
+
+// RsBwEcc implements Reed-Solomon error codes with Berlekamp-Welch error detection
+type RsBwEcc struct {
+	Options
+	FixedShareSize  bool
+	RequiredShares  int
+	RedundantShares int
+	Fec             *infectious.FEC
+}
+
+func NewRsBwEcc(opts *Options, fixedShareSize bool, spaceUsedPercentage float32) (*RsBwEcc, error) {
```

Comment:
also, Golang has slightly different convention for acronyms than say Java, so `ECC` would be all caps.

Perhaps function name would could be `NewReedSolomonBerlekampWelchECC` ? It's not going to be called in more than one or two places so the length does not matter much.